### DISCUSSION TITLE
Lesson 6 , File import callback not supported

### DISCUSSION THREAD

**Discussion by 0xalexhex** [Cyfrin/foundry-full-course-cu#32] [Q&A] [answered]
When I am trying to import Script from forge-std/Script.sol I am receiving this issue: Source "forge-std/Script.sol" not found: File import callback not supported. I tried to download the original Patric code from https://github.com/Cyfrin/foundry-simple-storage-f23 but have received the same issue.

**Comment by 0xconstructor** [comment]
you might be  using a version of the Solidity compiler  that doesn't support file import callbacks and Update Solidity Compiler ,check also absolute path of the file ur importing

**Reply by 0xalexhex** [reply]
That's the point I have the same issue even when I use Patrick code from link that he provided 
![Screenshot 2023-06-09 at 9 54 50 pm](https://github.com/Cyfrin/foundry-full-course-f23/assets/121734101/5ee5b2b8-e7f5-4667-8b39-faa031b8b959)

**Comment by cromewar** [comment, marked as answer]
Hello @AlexGorgut, does [this](https://ethereum.stackexchange.com/questions/142459/forge-std-test-sol-imported-and-working-but-vscode-marks-an-error) make sense for you?

**Reply by 0xalexhex** [reply]
Yes, thank you very much I wrote
```solidity
 import {Script} from "lib/forge-std/src/Script.sol";
// instead of  
import {Script} from "forge-std/Script.sol";
```
To be honest, when I just received this error, I tried to write it without a lib word, like this forge-std/src/Script.sol, but I thought that lib word has no difference :))

**Reply by cromewar** [reply]
I'm glad it helped!

**Comment by TreapGoGo** [comment]
I encountered the same issue and solved the problem. I've read various ways to solve this and some of them may be not the best. So let me make a summary.

1. @AlexGorgut tries to use
```solidity
import {Script} from "lib/forge-std/src/Script.sol";
// instead of  
import {Script} from "forge-std/Script.sol";
```
This really worked! But such a long path is not very beautiful. What's more, this method didn't solve the problem at all. It just avoids it with a complete path. And it doesn't make Patrick's code compiled.

The page provided by @cromewar mentions another two methods.

2. The first one tries to add `"solidity.remappings": ["forge-std/=lib/forge-std/src/",]` in `settings.json`. 
However, it doesn't work, at least not in my environment. Moreover, such a adjustment to settings may influence the global environment and cause unforeseen problems.

3. **[Best Solution]** The second one tries to run `forge remappings > remappings.txt` in the terminal. This command creates a `remapping.txt` which include `forge-std/=lib/forge-std/src/`, which is automatic translation from `forge-std` to `lib/forge-std/src`. 
This method saves the trouble to complete all the path and it won't affect other project too much. 
It works well in my environment. **IMPORTANT! If you find it invalid at first, try to close your VScode and relaunch.** The compiler will apply the `remappings.txt` and Patrick's code will be totally correct.

I hope it helps.

**Comment by sharonphiliplima** [comment]
Guys, I am still seeing this. 
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/69852010/af5e9519-0bc7-45ad-afee-06f240c5254f)

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/69852010/4f49426f-bbe6-4a3e-929e-8bf60a6ccdee)
